src/utils/readers/scientific.py

```
@@ -28,182 +46,269 @@
-                    attr_value = nc.getncattr(attr_name)
-                    lines.append(f"  - {attr_name}: {attr_value}")
-
-            text = "\n".join(lines)
-            logger.debug(f"Extracted metadata from NetCDF file {file_path.name}")
-            return text
-
+        with netCDF4.Dataset(path, "r") as nc:
+            return _parse_netcdf(nc, path.name)
     except Exception as e:  # Intentional catch-all: netCDF4 raises library-specific errors
-        raise FileReadError(f"Failed to read NetCDF file {file_path}: {e}") from e
+        raise FileReadError(f"Failed to read NetCDF file {path}: {e}") from e
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)